src/code/z_elf_message.c

```diff
@@ -1,60 +1,68 @@
 #include "global.h"
-#include "z64elf_message.h"
-
-ElfMessage sChildSariaMsgs[] = {
-    ELF_MSG_STRENGTH_UPG(SKIP, 3, false, 0),
-    ELF_MSG_FLAG(CHECK, 0x61, false, EVENTCHKINF_37),
-    ELF_MSG_END(0x64),
-    ELF_MSG_FLAG(CHECK, 0x62, false, EVENTCHKINF_25),
-    ELF_MSG_FLAG(CHECK, 0x63, false, EVENTCHKINF_37),
-    ELF_MSG_FLAG(CHECK, 0x65, false, EVENTCHKINF_43),
-    ELF_MSG_MEDALLION(CHECK, 0x66, false, ITEM_MEDALLION_FOREST),
-    ELF_MSG_MEDALLION(CHECK, 0x66, false, ITEM_MEDALLION_FIRE),
-    ELF_MSG_MEDALLION(CHECK, 0x66, false, ITEM_MEDALLION_WATER),
-    ELF_MSG_SONG(CHECK, 0x67, false, ITEM_SONG_STORMS),
-    ELF_MSG_MEDALLION(CHECK, 0x68, false, ITEM_MEDALLION_SPIRIT),
-    ELF_MSG_MEDALLION(CHECK, 0x68, false, ITEM_MEDALLION_SHADOW),
-    ELF_MSG_END(0x69),
+#include "z64hint_commands.h"
+
+HintCmd sChildSariaHints[] = {
+    HINT_CMD_STRENGTH_UPG(HINT_CMD_TYPE_SKIP, 0, false, 3),
+    HINT_CMD_FLAG(HINT_CMD_TYPE_CHECK, EVENTCHKINF_37, false, 0x61),
+    HINT_CMD_END(0x64),
+
+    HINT_CMD_FLAG(HINT_CMD_TYPE_CHECK, EVENTCHKINF_25, false, 0x62),
+    HINT_CMD_FLAG(HINT_CMD_TYPE_CHECK, EVENTCHKINF_37, false, 0x63),
+    HINT_CMD_FLAG(HINT_CMD_TYPE_CHECK, EVENTCHKINF_43, false, 0x65),
+    HINT_CMD_MEDALLION(HINT_CMD_TYPE_CHECK, ITEM_MEDALLION_FOREST, false, 0x66),
+    HINT_CMD_MEDALLION(HINT_CMD_TYPE_CHECK, ITEM_MEDALLION_FIRE, false, 0x66),
+    HINT_CMD_MEDALLION(HINT_CMD_TYPE_CHECK, ITEM_MEDALLION_WATER, false, 0x66),
+    HINT_CMD_SONG(HINT_CMD_TYPE_CHECK, ITEM_SONG_STORMS, false, 0x67),
+    HINT_CMD_MEDALLION(HINT_CMD_TYPE_CHECK, ITEM_MEDALLION_SPIRIT, false, 0x68),
+    HINT_CMD_MEDALLION(HINT_CMD_TYPE_CHECK, ITEM_MEDALLION_SHADOW, false, 0x68),
+    HINT_CMD_END(0x69),
 };
 
-ElfMessage sAdultSariaMsgs[] = {
-    ELF_MSG_MEDALLION(CHECK, 0x6A, false, ITEM_MEDALLION_FOREST),
-    ELF_MSG_MEDALLION(CHECK, 0x6B, false, ITEM_MEDALLION_FIRE),
-    ELF_MSG_MEDALLION(CHECK, 0x6B, false, ITEM_MEDALLION_WATER),
-    ELF_MSG_MEDALLION(CHECK, 0x6C, false, ITEM_MEDALLION_SPIRIT),
-    ELF_MSG_MEDALLION(CHECK, 0x6C, false, ITEM_MEDALLION_SHADOW),
-    ELF_MSG_END(0x6D),
+HintCmd sAdultSariaHints[] = {
+    HINT_CMD_MEDALLION(HINT_CMD_TYPE_CHECK, ITEM_MEDALLION_FOREST, false, 0x6A),
+    HINT_CMD_MEDALLION(HINT_CMD_TYPE_CHECK, ITEM_MEDALLION_FIRE, false, 0x6B),
+    HINT_CMD_MEDALLION(HINT_CMD_TYPE_CHECK, ITEM_MEDALLION_WATER, false, 0x6B),
+    HINT_CMD_MEDALLION(HINT_CMD_TYPE_CHECK, ITEM_MEDALLION_SPIRIT, false, 0x6C),
+    HINT_CMD_MEDALLION(HINT_CMD_TYPE_CHECK, ITEM_MEDALLION_SHADOW, false, 0x6C),
+    HINT_CMD_END(0x6D),
 };
 
-u32 ElfMessage_CheckCondition(ElfMessage* msg) {
-    s32 type = msg->byte0 & 0x1E;
+u32 Hints_CheckCondition(HintCmd* hints) {
+    s32 type = hints->byte0 & 0x1E;
```

Comment:
Macros relating these masks/shifts to the byte packing macros in `z64hint_commands.h` may make it easier for someone to edit the system?
Not sure if it's realistic however as there's a lot to cover here

Reply:
I agree macros should be made, but Im not exactly sure how id approach it. If someone else wants to I think itd be good.
But also, this system is kinda ass and anyone modding is probably going to rip it out like mm did, so Im not the most concerned about usability from that point of view lmao. But from a documentation standpoint, yea agree